src/app/qgssettingstree.cpp

```diff
@@ -319,7 +321,17 @@ QTreeWidgetItem *QgsSettingsTree::createItem( const QString &text,
     item->setFlags( item->flags() | Qt::ItemIsEditable );
 
   item->setData( 0, TypeRole, isGroup ? Group : Setting );
-  item->setData( 0, PathRole, mSettings->group().isEmpty() ? text : mSettings->group() + '/' + text );
+
+  QString completeSettingsPath = mSettings->group().isEmpty() ? text : mSettings->group() + '/' + text;
+  item->setData( 0, PathRole, completeSettingsPath );
+
+  // If settings registered add description
+  if ( !isGroup )
+  {
+    const QgsSettingsEntryBase *settingsEntry = QgsApplication::settingsRegistryCore()->getSettingsEntry( completeSettingsPath, true );
+    if ( settingsEntry != nullptr )
+      item->setText( 3, settingsEntry->description() );
```

Comment:
What is this `3` ? If it's a column identifier you must create an enum with meaningful names.

Reply:
I was not comfortable writing 3. But the rest of the class uses literals everywhere for the columns so I have just done the same. I'll change it.

Reply:
Thank you, it happens all the times: when you touch old code you make it better :smiley_cat:

Reply:
nice i rather like that as "don't touch if it's working". It's now committed

Reply:
Yeah, I thought that too before being asked thousands of times to change some parts of the code that I didn't touch or write. 

But if everyone of us tries to leave the code in a better shape it ultimately leads to a better and more readable code, coding conventions have changed a lot during the past decades, what was once acceptable it isn't anymore and we have a chance to modernize the code while working on it.